Crashes due to network errors

Execution failed probably due to network timeouts.

```
2021-01-26T13:18:52.9260303Z bundler: failed to load command: iev-termbase (/opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/bin/iev-termbase)
2021-01-26T13:18:52.9263764Z /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/protocol.rb:217:in `rbuf_fill': Net::ReadTimeout with #<TCPSocket:(closed)> (Net::ReadTimeout)
```

See logs. Not very important but quite annoying.

<details>
<summary>Logs</summary>

```
2021-01-26T13:16:50.7934186Z Processing term 802-01-01 (ara)... 
2021-01-26T13:16:50.7935332Z Processing term 802-01-01 (deu)... 
2021-01-26T13:16:50.7936524Z Processing term 802-01-01 (eng)... 
2021-01-26T13:16:50.7937686Z Processing term 802-01-01 (spa)... 
2021-01-26T13:16:50.7938838Z Processing term 802-01-01 (fra)... 
2021-01-26T13:16:50.7940017Z Processing term 802-01-01 (jpn)... 
2021-01-26T13:16:50.7941184Z Processing term 802-01-01 (kor)... 
2021-01-26T13:16:50.7942360Z Processing term 802-01-01 (pol)... 
2021-01-26T13:16:50.7943515Z Processing term 802-01-01 (por)... 
2021-01-26T13:16:50.7944696Z Processing term 802-01-01 (srp)... 
2021-01-26T13:16:50.7945849Z Processing term 802-01-01 (zho)... 
2021-01-26T13:16:50.7946996Z Processing term 802-01-02 (ara)... 
2021-01-26T13:16:50.7948183Z Processing term 802-01-02 (deu)... 
2021-01-26T13:16:50.7949344Z Processing term 802-01-02 (eng)... 
2021-01-26T13:16:50.7950528Z Processing term 802-01-02 (spa)... 
2021-01-26T13:16:50.7951678Z Processing term 802-01-02 (fra)... 
2021-01-26T13:16:50.7952848Z Processing term 802-01-02 (jpn)... 
2021-01-26T13:16:50.7954008Z Processing term 802-01-02 (kor)... 
2021-01-26T13:16:50.7955172Z Processing term 802-01-02 (pol)... 
2021-01-26T13:16:50.7956353Z Processing term 802-01-02 (por)... 
2021-01-26T13:16:50.7957524Z Processing term 802-01-02 (srp)... 
2021-01-26T13:16:50.7959615Z Processing term 802-01-02 (zho)... 
2021-01-26T13:16:50.7961348Z Processing term 802-01-03 (ara)... 
2021-01-26T13:16:50.7963045Z Processing term 802-01-03 (deu)... 
2021-01-26T13:16:50.7964263Z Processing term 802-01-03 (eng)... 
2021-01-26T13:16:50.7965438Z Processing term 802-01-03 (spa)... 
2021-01-26T13:16:50.7966629Z Processing term 802-01-03 (fra)... 
2021-01-26T13:16:50.7968149Z Processing term 802-01-03 (jpn)... 
2021-01-26T13:16:50.7969334Z Processing term 802-01-03 (kor)... 
2021-01-26T13:16:50.7970505Z Processing term 802-01-03 (pol)... 
2021-01-26T13:16:50.7971702Z Processing term 802-01-03 (por)... 
2021-01-26T13:16:50.7972873Z Processing term 802-01-03 (srp)... 
2021-01-26T13:16:50.7974041Z Processing term 802-01-03 (zho)... 
2021-01-26T13:16:50.7975216Z Processing term 802-01-04 (ara)... 
2021-01-26T13:16:50.8223036Z [relaton-iec] ("IEC 62127-1") fetching...
2021-01-26T13:18:52.9260303Z bundler: failed to load command: iev-termbase (/opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/bin/iev-termbase)
2021-01-26T13:18:52.9263764Z /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/protocol.rb:217:in `rbuf_fill': Net::ReadTimeout with #<TCPSocket:(closed)> (Net::ReadTimeout)
2021-01-26T13:18:52.9265420Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/protocol.rb:191:in `readuntil'
2021-01-26T13:18:52.9266747Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/protocol.rb:201:in `readline'
2021-01-26T13:18:52.9268145Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http/response.rb:40:in `read_status_line'
2021-01-26T13:18:52.9270700Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http/response.rb:29:in `read_new'
2021-01-26T13:18:52.9271945Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:1509:in `block in transport_request'
2021-01-26T13:18:52.9273089Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:1506:in `catch'
2021-01-26T13:18:52.9274227Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:1506:in `transport_request'
2021-01-26T13:18:52.9275367Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:1479:in `request'
2021-01-26T13:18:52.9276461Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:1380:in `request_get'
2021-01-26T13:18:52.9277629Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:483:in `block in get_response'
2021-01-26T13:18:52.9278746Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:920:in `start'
2021-01-26T13:18:52.9279818Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:605:in `start'
2021-01-26T13:18:52.9280913Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/2.6.0/net/http.rb:481:in `get_response'
2021-01-26T13:18:52.9282550Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/scrapper.rb:229:in `fetch_status_relations'
2021-01-26T13:18:52.9285666Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/scrapper.rb:42:in `parse_page'
2021-01-26T13:18:52.9287192Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/hit.rb:9:in `fetch'
2021-01-26T13:18:52.9288775Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/iec_bibliography.rb:182:in `results_filter'
2021-01-26T13:18:52.9290475Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/iec_bibliography.rb:200:in `iecbib_get'
2021-01-26T13:18:52.9292112Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/iec_bibliography.rb:46:in `get'
2021-01-26T13:18:52.9293691Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-iec-1.7.5/lib/relaton_iec/processor.rb:17:in `get'
2021-01-26T13:18:52.9295138Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-1.7.2/lib/relaton/db.rb:204:in `new_bib_entry'
2021-01-26T13:18:52.9296569Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-1.7.2/lib/relaton/db.rb:191:in `check_bibliocache'
2021-01-26T13:18:52.9297951Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/relaton-1.7.2/lib/relaton/db.rb:34:in `fetch'
2021-01-26T13:18:52.9299177Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/relaton_db.rb:16:in `fetch'
2021-01-26T13:18:52.9300634Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/source_parser.rb:286:in `extract_single_source'
2021-01-26T13:18:52.9301929Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/source_parser.rb:18:in `block in parse'
2021-01-26T13:18:52.9303121Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/source_parser.rb:18:in `map'
2021-01-26T13:18:52.9304307Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/source_parser.rb:18:in `parse'
2021-01-26T13:18:52.9305518Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/source_parser.rb:11:in `initialize'
2021-01-26T13:18:52.9306812Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/term_builder.rb:359:in `new'
2021-01-26T13:18:52.9308135Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/term_builder.rb:359:in `extract_authoritative_source'
2021-01-26T13:18:52.9309505Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/term_builder.rb:65:in `build_term_object'
2021-01-26T13:18:52.9310719Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/term_builder.rb:16:in `build'
2021-01-26T13:18:52.9311912Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/term_builder.rb:20:in `build_from'
2021-01-26T13:18:52.9315685Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/concept_collection.rb:15:in `block (2 levels) in build_from_dataset'
2021-01-26T13:18:52.9318500Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/dataset/actions.rb:152:in `block in each'
2021-01-26T13:18:52.9321363Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:345:in `block (2 levels) in fetch_rows'
2021-01-26T13:18:52.9324421Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sqlite3-1.4.2/lib/sqlite3/resultset.rb:134:in `each'
2021-01-26T13:18:52.9327081Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:334:in `block in fetch_rows'
2021-01-26T13:18:52.9329719Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sqlite3-1.4.2/lib/sqlite3/database.rb:339:in `query'
2021-01-26T13:18:52.9332449Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:200:in `block (2 levels) in _execute'
2021-01-26T13:18:52.9335314Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/database/logging.rb:38:in `log_connection_yield'
2021-01-26T13:18:52.9338119Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:200:in `block in _execute'
2021-01-26T13:18:52.9340880Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/connection_pool/threaded.rb:92:in `hold'
2021-01-26T13:18:52.9343679Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/database/connecting.rb:269:in `synchronize'
2021-01-26T13:18:52.9346462Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:193:in `_execute'
2021-01-26T13:18:52.9349114Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:139:in `execute'
2021-01-26T13:18:52.9351764Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/dataset/actions.rb:1093:in `execute'
2021-01-26T13:18:52.9354414Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/adapters/sqlite.rb:327:in `fetch_rows'
2021-01-26T13:18:52.9357050Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/sequel-5.40.0/lib/sequel/dataset/actions.rb:152:in `each'
2021-01-26T13:18:52.9359650Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/concept_collection.rb:14:in `block in build_from_dataset'
2021-01-26T13:18:52.9360977Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/concept_collection.rb:13:in `tap'
2021-01-26T13:18:52.9364496Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/concept_collection.rb:13:in `build_from_dataset'
2021-01-26T13:18:52.9365861Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/cli/command.rb:10:in `xlsx2yaml'
2021-01-26T13:18:52.9367127Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
2021-01-26T13:18:52.9368492Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
2021-01-26T13:18:52.9369966Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
2021-01-26T13:18:52.9371235Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
2021-01-26T13:18:52.9372381Z  from /home/runner/work/iev-data/iev-data/lib/iev/termbase/cli.rb:5:in `start'
2021-01-26T13:18:52.9373486Z  from /home/runner/work/iev-data/iev-data/exe/iev-termbase:21:in `<top (required)>'
2021-01-26T13:18:52.9375033Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/bin/iev-termbase:23:in `load'
2021-01-26T13:18:52.9376255Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/bin/iev-termbase:23:in `<top (required)>'
2021-01-26T13:18:52.9377594Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/cli/exec.rb:63:in `load'
2021-01-26T13:18:52.9379000Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/cli/exec.rb:63:in `kernel_load'
2021-01-26T13:18:52.9380404Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/cli/exec.rb:28:in `run'
2021-01-26T13:18:52.9383910Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/cli.rb:494:in `exec'
2021-01-26T13:18:52.9385369Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
2021-01-26T13:18:52.9387003Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
2021-01-26T13:18:52.9388611Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
2021-01-26T13:18:52.9390345Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/cli.rb:30:in `dispatch'
2021-01-26T13:18:52.9391848Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
2021-01-26T13:18:52.9393258Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/cli.rb:24:in `start'
2021-01-26T13:18:52.9394659Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/exe/bundle:49:in `block in <top (required)>'
2021-01-26T13:18:52.9396182Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/lib/bundler/friendly_errors.rb:130:in `with_friendly_errors'
2021-01-26T13:18:52.9397667Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/bundler-2.2.6/exe/bundle:37:in `<top (required)>'
2021-01-26T13:18:52.9398801Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle:23:in `load'
2021-01-26T13:18:52.9399771Z  from /opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle:23:in `<main>'
2021-01-26T13:18:52.9605367Z Processing term 802-01-04 (deu)... 
2021-01-26T13:18:52.9606133Z Processing term 802-01-04 (eng)... 
2021-01-26T13:18:52.9606841Z Processing term 802-01-04 (spa)... 
2021-01-26T13:18:52.9607513Z Processing term 802-01-04 (fra)... 
2021-01-26T13:18:52.9608166Z Processing term 802-01-04 (jpn)... 
2021-01-26T13:18:52.9608832Z Processing term 802-01-04 (kor)... 
2021-01-26T13:18:52.9609483Z Processing term 802-01-04 (pol)... 
2021-01-26T13:18:52.9610139Z Processing term 802-01-04 (por)... 
2021-01-26T13:18:52.9611086Z Processing term 802-01-04 (srp)... 
2021-01-26T13:18:52.9611791Z Processing term 802-01-04 (zho)... 
2021-01-26T13:18:52.9612461Z Processing term 802-01-05 (ara)... 
2021-01-26T13:18:52.9613112Z Processing term 802-01-05 (deu)... 
2021-01-26T13:18:52.9613776Z Processing term 802-01-05 (eng)... 
2021-01-26T13:18:52.9614429Z Processing term 802-01-05 (spa)... 
2021-01-26T13:18:52.9615076Z Processing term 802-01-05 (fra)... 
2021-01-26T13:18:52.9615738Z Processing term 802-01-05 (jpn)... 
2021-01-26T13:18:52.9616387Z Processing term 802-01-05 (kor)... 
2021-01-26T13:18:52.9617051Z Processing term 802-01-05 (pol)... 
2021-01-26T13:18:52.9617787Z Processing term 802-01-05 (por)... 
2021-01-26T13:18:52.9618468Z Processing term 802-01-05 (srp)... 
2021-01-26T13:18:52.9619117Z Processing term 802-01-05 (zho)... 
2021-01-26T13:18:52.9619777Z Processing term 802-01-06 (ara)... 
2021-01-26T13:18:52.9620425Z Processing term 802-01-06 (deu)... 
2021-01-26T13:18:52.9803161Z Processing term 802-01-06 (eng)... Makefile:19: recipe for target 'termbase.yaml' failed
2021-01-26T13:18:52.9804103Z make: *** [termbase.yaml] Error 1
2021-01-26T13:18:52.9814233Z ##[error]Process completed with exit code 2.
2021-01-26T13:18:52.9928166Z Post job cleanup.
2021-01-26T13:18:53.1177616Z [command]/usr/bin/git version
2021-01-26T13:18:53.1241687Z git version 2.29.2
2021-01-26T13:18:53.1290614Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2021-01-26T13:18:53.1347217Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2021-01-26T13:18:53.1679839Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2021-01-26T13:18:53.1712014Z http.https://github.com/.extraheader
2021-01-26T13:18:53.1724363Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2021-01-26T13:18:53.1775890Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2021-01-26T13:18:53.2099606Z Cleaning up orphan processes
```

</details>

--- a/lib/iev/termbase/relaton_db.rb
+++ b/lib/iev/termbase/relaton_db.rb
@@ -13,7 +13,28 @@ module Iev
       # @param code [String] reference
       # @return [RelatonIso::IsoBibliongraphicItem]
       def fetch(code)
-        @db.fetch code
+        retrying_on_failures do
+          @db.fetch code
+        end
+      end
+
+      private
+
+      def retrying_on_failures(attempts: 4)
+        curr_attempt = 1
+
+        begin
+          yield
+
+        rescue
+          if curr_attempt <= attempts
+            sleep(2 ** curr_attempt * 0.1)
+            curr_attempt += 1
+            retry
+          else
+            raise
+          end
+        end
       end
     end
   end

